### ISSUE TITLE
Apply all safe ty fixes in a Python file

### ISSUE THREAD

**Issue by lerebear** [astral-sh/ty#4136]
## Problem

Users must currently apply individual ty quick fixes one at a time. When a file contains several redundant casts, unused suppressions, or other safely fixable diagnostics, this becomes repetitive and cannot be integrated cleanly with editor save actions.

For example:

```python
from typing import cast

first = cast(int, 1)
second = cast(str, "value")
```

A fix-all action should remove both unnecessary casts in one operation.

By constrast, [Pyrefly supports file-wide `source.fixAll` actions](https://github.com/facebook/pyrefly/blob/b52e609470554983d33a2cb84074f619d5272d85/pyrefly/lib/lsp/non_wasm/server.rs#L1356).

## Candidate fix-all operations

- [ ] Remove redundant casts.
- [ ] Remove unused diagnostic suppressions.
- [ ] Apply safe diagnostic-provided replacements.
- [ ] Add unambiguous missing imports.
- [ ] Resolve overlapping edits deterministically.
- [ ] Optionally support narrowly scoped rule-specific fix-all actions.

## Acceptance criteria

- [ ] Editors can request a `source.fixAll` action that applies all compatible safe ty fixes in the current file.
- [ ] Applying the action combines non-overlapping fixes without introducing invalid Python or changing unrelated code.
- [ ] Running the action again without further changes produces no additional edits.
